### PR TITLE
Add missing highlighting for Python snippets

### PR DESCRIPTION
**Pull request by aslisabanci**
## What does this PR do?
Adds missing highlighting for Python for example and test code snippets. Can continue with similar fixes for the remaining doc pages if this PR looks okay.

### Does your PR introduce any breaking changes ? If yes, please list them.
No. 

## Before submitting
- [N/A] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [N/A] Did you make sure to update the documentation with your changes? (if necessary)
- [N/A] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [N/A] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)
- [x] Did you list all the breaking changes introduced by this pull request?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun? 
YES, excited for the 1st PR!🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]

File: docs/source/common/trainer.rst [hunk 1]
```diff
@@ -266,13 +272,15 @@ accumulate_grad_batches
 Accumulates grads every k batches or as set up in the dict.
 Trainer also calls ``optimizer.step()`` for the last indivisible step number.
 
-.. testcode::
+.. testcode:: python
```
Comment (Borda) [inline docs/source/common/trainer.rst:275]: we prefer `testcode` if possible

**Review by Borda — CHANGES_REQUESTED** [review]

File: docs/source/common/trainer.rst [hunk 2]
```diff
@@ -182,6 +182,8 @@ and set ``deterministic`` flag in ``Trainer``.
 
 Example::
 
+.. code-block:: python
+
```
Comment (awaelchli) [inline docs/source/common/trainer.rst:186]: you cannot put this inside inside the example block, or it will become code. 
please check the output:

image

you can generate the docs like so:
- pip install -r requirements/docs.txt
- cd docs
- make html 
and then browse to docs/build and open the index.html file for example. then navigate to trainer.

**Review by awaelchli — CHANGES_REQUESTED** [review]

File: docs/source/common/trainer.rst (hunk 2 above)
Comment (awaelchli) [reply docs/source/common/trainer.rst:186]: If you need help, let me know. 
Happy to assist @aslisabanci

Comment (aslisabanci) [reply docs/source/common/trainer.rst]: Ahh you're right! I could've even caught this from Github's preview - but of course I'll make the docs with your instructions and double check them this way from now on. 
For a quick fix I just removed the code-block sections from under Examples and sent a new commit, but for my next PRs I'll make sure to do this beforehand. This one was in rush to be the 500th contributor :)

If I need any assistance making the docs I'll reach out to you too, thanks a bunch for your support Adrian!

File: docs/source/common/trainer.rst (hunk 1 above)
Comment (aslisabanci) [reply docs/source/common/trainer.rst]: I think I saw this usage in `optimizers.rst` so I wanted to follow the same but it's alright if we prefer just `testcode` In my next commit I reverted those. Thank you for letting me know!

File: docs/source/common/trainer.rst [hunk 3]
```diff
@@ -225,6 +225,8 @@ The accelerator backend to use (previously known as distributed_backend).
 
 .. testcode::
 
+.. code-block:: python
+
```
Comment (Borda) [inline docs/source/common/trainer.rst:229]:
```suggestion
```

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
thanks for the help here @aslisabanci <3
